controllers/grafanafolder_controller.go

```diff
@@ -328,10 +328,14 @@ func (r *GrafanaFolderReconciler) onFolderCreated(ctx context.Context, grafana *
 			}
 		}
 	} else {
-		folderResp, err := grafanaClient.Folders.CreateFolder(&models.CreateFolderCommand{
+		body := &models.CreateFolderCommand{
 			Title: title,
 			UID:   uid,
-		})
+		}
+		if cr.HasAParentFolder() {
+			body.ParentUID = cr.Spec.ParentFolderUID
+		}
+		folderResp, err := grafanaClient.Folders.CreateFolder(body)
```

Comment:
What I've also noticed is that folder [move API](https://grafana.com/docs/grafana/latest/developers/http_api/folder/#move-folder) is not implemented, so if someone changes the CR, it will not be reflected in a Grafana instance. It'd be nice to have it.

Reply:
Ok, I'll investigate to find a way to do it.

Reply:
I think I have a first version of this one. I got difficulties with the reconciliation loop. However, now I have something working.

I let you check if everything looks fine for you. :)

Reply:
@weisdd Do you take a look at this change?

If you want more information about it, you can find it [here](https://github.com/grafana/grafana-operator/pull/1566/files#diff-096a42dca40b047281babb3a3dd27a44532200616613e71f95e2ab04c29133bcR332).